crates/moon/src/cli/new.rs

```diff
@@ -88,105 +63,43 @@ pub fn run_new(_cli: &UniversalFlags, cmd: NewSubcommand) -> anyhow::Result<i32>
-
-                    let p = PathBuf::from(p);
-
-                    if p.exists() {
-                        if p.join(MOON_MOD_JSON).exists() {
-                            Err(format!(
-                                "A MoonBit project already exists in `{}`.",
-                                p.display()
-                            ))
-                        } else {
-                            if !dot {
-                                eprintln!(
-                                    "{}: The directory is already exists.",
-                                    "Warning".yellow().bold(),
-                                );
-                            };
-                            Ok(())
-                        }
-                    } else {
-                        Ok(())
-                    }
-                })
-                .interact()?;
-            let path = PathBuf::from(tmp);
-
-            let existing_username = get_existing_username();
-            let default_username = existing_username.unwrap_or_else(|| "username".to_string());
-
-            let username = dialoguer::Input::<String>::new()
-                .with_prompt("Enter your username")
-                .default(default_username)
-                .validate_with(|input: &String| -> Result<(), String> { validate_username(input) })
-                .show_default(true)
-                .interact()?;
-
-            let package_name = dialoguer::Input::<String>::new()
-                .with_prompt("Enter your project name")
-                .default("hello".to_string())
-                .show_default(true)
-                .interact()?;
-
-            let license = dialoguer::Input::<String>::new()
-                .with_prompt("Enter your license")
-                .default("Apache-2.0".to_string())
-                .show_default(true)
-                .interact()?;
-
-            (path, username, package_name, Some(license))
-        };
-
-    if target_dir.exists() && target_dir.join(MOON_MOD_JSON).exists() {
-        bail!(
-            "A MoonBit project already exists in `{}`.",
-            target_dir.display()
-        );
+    let existing_username = get_existing_username();
+    let username = cmd
+        .user
+        .or(existing_username)
+        .unwrap_or_else(|| "username".to_string());
+    validate_username(&username).map_err(|e| anyhow::anyhow!(e))?;
+
+    let project_name = match cmd.name {
+        Some(name) => name,
+        None => {
+            let absolute_path = if path.is_relative() {
+                match std::env::current_dir() {
+                    Ok(current_dir) => current_dir.join(&path),
+                    Err(_) => path.clone(),
+                }
+            } else {
+                path.clone()
+            };
+            absolute_path
+                .file_stem()
+                .and_then(|s| s.to_str())
+                .map(|s| s.to_string())
+                .unwrap_or_else(|| "hello".to_string())
```

Comment:
Nit: If you care about memory efficiency you can use `Cow<Path>` here; but this is merely just nitpicking. Using a `clone` or two won't do any harm to a run-once script.